### PR TITLE
Update Flax NNX Randomness

### PR DESCRIPTION
**Pull request by 8bitmp3**
@cgarciae @IvyZX PTAL

Preview: https://flax--4279.org.readthedocs.build/en/4279/guides/randomness.html

### CODE REVIEW

**Comment by 8bitmp3** [comment]
Updating:

```diff
- Here is a list of the main PRNG-related types in Flax NNX:
+ Here are the main PRNG-related types in Flax NNX:
```

Added the JAX `jax.random` tutorial for new users:

```
> **Note:** To learn more about random number generation in JAX,
the `jax.random` API, and PRNG-generated sequences, check out
this [JAX PRNG tutorial](https://jax.readthedocs.io/en/latest/random-numbers.html).
```
Added `nnx.`:

```
...`nnx.Rngs` object..... `nnx.reseed`.... `nnx.Dropout`
```

**Comment by 8bitmp3** [comment]
rebasing after https://github.com/google/flax/pull/4281

**Comment by 8bitmp3** [comment]
@cgarciae PTAL, thanks!

File: docs_nnx/guides/randomness.ipynb [hunk 1]
```diff
@@ -318,8 +344,14 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Splitting Rngs\n",
-    "When interacting with transforms like `vmap` or `pmap` it is often necessary to split the random state such that each replica has its own unique state. This can be done in two way, either by manually splitting a key before passing it to one of the `Rngs` streams, or using the `nnx.split_rngs` decorator which will automatically split the random state of any `RngStream`s found in the inputs of the function and automatically \"lower\" them once the function call ends. `split_rngs` is more convenient as it works nicely with transforms so we'll show an example of that here:"
+    "## Splitting `nnx.Rngs`\n",
```
Comment (IvyZX) [inline docs_nnx/guides/randomness.ipynb:347]: This is not splitting `nnx.Rngs` directly. Maybe say "Splitting RNG keys".

**Review by IvyZX — APPROVED** [review]

File: docs_nnx/guides/randomness.ipynb (hunk 1 above)
Comment (8bitmp3) [reply docs_nnx/guides/randomness.ipynb:347]:
```
## Splitting PRNG keys
```

thank you @IvyZX

**Pull request merged**